### DISCUSSION TITLE
Other Arduino Board Support

### DISCUSSION THREAD

**Discussion by daba1208** [lcgamboa/picsimlab#66] [Q&A] [answered]
I downloaded and installed PICSimLab hoping I could use it with (firstly) Arduino Nano, and (secondly) with Lolin (Wemos) D1 mini.

I'm guessing I can use the Uno board for the Nano, but it would be great if the software supported it 100%.

I tried to follow the "Creating New Boards" but it all got way too much for me, and I'm thinking it would be relatively trivial for "experts" to do this without stalling at each step because of undocumented unknowns.

I'm a complete noob with GitHub, and see so much it scares me.

A step-by-step guide (more verbose than in PICSimLab) would be a godsend ....

**Comment by lcgamboa** [comment, marked as answer]
The Arduino Nano board has the same hardware and pinout as the Arduino Uno, only the board format is smaller. One can replace the other without the need for firmware changes.
As my time to maintain PICSimLab is short, I consider it redundant to have both boards. But basically, just duplicate the Arduino Uno source code and change the board picture to add nano support. (And two duplicated source codes to keep :( )

The D1 mini is based on ESP8266, I don't know if exists any simulator for it. Making one from scratch is no simple task. Not even Espressif recommends its use in new projects anymore and it is not integrated into the IDF. I don't think it's worth the effort.

The "Creating New Boards" tutorial is for already supported microcontrollers, as in the case of the ATMega328 of the Nano, and not applied to the case of the ESP8266. But this tutorial requires a fair knowledge of programming.

Another example is the ATmega2560 which is only supported on the generic Breadboard board but runs Arduino Mega codes without any problems.

**Comment by lcgamboa** [comment]
Hi @daba1208,

I added the Arduino Nano board in the [last commit](https://github.com/lcgamboa/picsimlab/commit/7d2d431f78903097f0b238caeb76a87c0a431507#diff-e1c03cd4ff00df51dde3f5088cf1fcab1761dc0b38d3b0cd3a1aeecdd9653388).
You can test using the [unstable version](https://github.com/lcgamboa/picsimlab/releases/download/latestbuild/PICSimLab_0.8.12_221022_win64_setup.exe).

![Peek 23-10-2022 10-31](https://user-images.githubusercontent.com/13749258/197395298-52383257-7213-4097-abdf-9929409d565c.gif)

As for the Lolin (Wemos) D1 mini for the reasons I already mentioned there will probably be no implementation by me.

**Comment by daba1208** [comment]
Great stuff ! I’ll take a good look when I get the chance. 

Both my current projects (Nano and Wemos mini), have been completed and working to my specifications …

I might be on a Uno next, as I want to build a DMX controller, and I already have the Uno DMX shield.
